### ISSUE TITLE
Improve handling of compiler package

### ISSUE THREAD

**Issue by revodavid** [RevolutionAnalytics/checkpoint#159]
Follow-up to #156 

The current implementation installs a copy of the "compiler" package in each .checkpoint folder, which takes time and uses disk space.

Can we avoid that install and instead make "compiler" available on the library search path when checkpoint is active?

**Comment by andrie** [comment]
Near duplicate of #156

**Comment by andrie** [comment]
This was a workaround to deal with the vagaries of how `compiler` gets installed in R.  By exposing the `compiler` search path, we also risk exposing other packages, thus risking non-reproducible behaviour of `checkpoint`.

So I think we should not modify this behaviour.

However, we can check if compiler is already in the `.checkpoint` search path and not re-install multiple times.

**Comment by piccolbo** [comment]
We could have a special compiler directory, together with .checkpoint/date1, .checkpoint/date2 etc we could have a .checkpoint/base or some such that doesn't change by date.  After a checkpoint the search path would be 
c(".checkpoint/somedate", ".checkpoint/base")
or some such. Of course it adds complexity but if we are really concerned about space the current method installs compiler once for each date. +1 for checking if installing is necessary, but that doesn't spare one copy per checkpoint. With the interplay of two dedicated checkpoint directories, under exclusive control of checkpoint, we can both keep the reproducibility promise and install compiler only once. What do you think?

**Comment by revodavid** [comment]
To answer that question, I need to understand why the "compiler" package needs special treatment when other base packages don't. Can you explain?

**Comment by andrie** [comment]
From the comments in https://github.com/RevolutionAnalytics/checkpoint/issues/143 on Nov 26:

```r
> system.file(package = "compiler")
[1] "/Library/Frameworks/R.framework/Versions/3.1/Resources/library/compiler"
> system.file(package = "base")
[1] "/Library/Frameworks/R.framework/Resources/library/base"
> system.file(package = "stats4")
[1] "/Library/Frameworks/R.framework/Resources/library/stats4"
> system.file(package = "graphics")
[1] "/Library/Frameworks/R.framework/Resources/library/graphics"
```

**Comment by revodavid** [comment]
Odd. On my machine at least, compiler is also in /Library/Frameworks/R.framework/Resources/library. In fact it seems all of the base packages (base, stats, utils etc) are in both locations.

**Comment by piccolbo** [comment]
Yes, I don't have a very good explanation either other than what @andrie quoted. Only compiler needs this treatment because only compiler seems to be affected by this problem. Tried every other base package one by one. Knowing the root cause would be  nice, by the payoff, other than a fulfilled intellectual curiosity, would be limited.

**Comment by andrie** [comment]
I'm working on a solution that will install compiler (and any other missing base packages) in a single, shared folder per R version, e.g `~/.checkpoint/R-3.1.2`

Push to dev following shortly.
